Signal/src/util/NSTimer+OWS.m

```diff
@@ -0,0 +1,65 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "NSTimer+OWS.h"
+#import <objc/runtime.h>
+
+@interface NSTimerProxy : NSObject
+
+@property (nonatomic, weak) id target;
+@property (nonatomic) SEL selector;
+
+@end
+
+#pragma mark -
+
+@implementation NSTimerProxy
+
+- (void)timerFired:(NSDictionary *)userInfo
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+    [self.target performSelector:self.selector withObject:userInfo];
+#pragma clang diagnostic pop
+}
+
+@end
+
+#pragma mark -
+
+static void *kNSTimer_OWS_Proxy = &kNSTimer_OWS_Proxy;
+
+@implementation NSTimer (OWS)
+
+- (NSTimerProxy *)proxy
```

Comment:
nit: to avoid shooting yourself in the foot, especially with short method names, it's good to prefix category methods, e.g. `ows_proxy`